lib/base-controller.js

```diff
@@ -26,27 +26,37 @@ function getErrorLength() {
 BaseController.prototype.getNextStep = function getNextStep(req, res) {
   var next = Controller.prototype.getNextStep.apply(this, arguments);
   var forks = (this.options || {}).forks || [];
+  var forked = false;
 
   _.each(forks, function eachFork(fork) {
     if (_.isFunction(fork.condition)) {
       if (fork.condition(req, res)) {
-        next = req.baseUrl + fork.target;
+        if (!_.includes(req.sessionModel.get('steps'), fork.target)) {
+          next = req.baseUrl + fork.target;
+          forked = true;
+        }
       }
     }
     if (_.isPlainObject(fork.condition)) {
       if (fork.condition.value === req.form.values[fork.condition.field]) {
-        next = req.baseUrl + fork.target;
+        if (!_.includes(req.sessionModel.get('steps'), fork.target)) {
+          next = req.baseUrl + fork.target;
+          forked = true;
+        }
       }
     }
   });
 
-  if (req.params.action === 'edit' && !this.options.continueOnEdit) {
+  if (forked) {
+    return next;
```

Comment:
Git blame says no you dont

https://github.com/UKHomeOffice/hof-controllers/blame/cb47acd758962ee6ab82636399f944f044f983f8/lib/base-controller.js#L31

👯 👯 👯 👯 👯 👯